forem/forem
# fix: Clear chat input after Enter key submit

## Description

The chat input value was handled and reset outside of the component. While that is not necessarily a problem, the auto resized Textarea component acted weird when the value was not "controlled" by the `Textarea` component.

The solution was indeed to hold a state inside the compose component, but also to make sure the only place where the input of the textarea is controlled was inside the component itself, and nowhere else. 

Also in the PR:
* Compose becomes a functional component, which made it easier to manage this particular state and props flow
* Use 1 textarea instance for both add and edit mode

## Related Tickets & Documents

Closes #10480 

## QA Instructions, Screenshots, Recordings

* Open chat
* Make sure the input is cleared when sent with the Enter key
* Make sure the input is cleared when sent with the send button
* Make sure the textarea is resized when the input overflows

All of the above should be true in edit mode as well.

## Added tests?


The tests check for an empty textarea field after the following events:

* Pressing Enter
* Clicking Send
* Clicking Save edit
* Clicking Close edit

## Added to documentation?
